f8a_report/retrain_main.py

```diff
@@ -0,0 +1,27 @@
+"""Retraining pipeline initiation."""
+import logging
+from datetime import datetime as dt, timedelta
+from report_helper import ReportHelper
+
+logger = logging.getLogger(__file__)
+
+
+def main():
+    """Retraining pipeline initiation."""
+    r = ReportHelper()
+    today = dt.today()
+
+    # Weekly re-training of models
+    start_date_wk = (today - timedelta(days=7)).strftime('%Y-%m-%d')
+    end_date_wk = today.strftime('%Y-%m-%d')
+
+    logger.info('Weekly Job Triggered')
+    try:
+        r.re_train(start_date_wk, end_date_wk, 'weekly', retrain=True)
+    except Exception as e:
+        logger.exception("Exception in Retraining")
+        raise e
```

Comment:
Should we really handle the exception here? I prefer to propagate it.

Reply:
We are raising the exception. This helps in documentation, people who look at the code know that there is a possibility of an exception. Also helps in custom logging.